taskt/Core/Automation/Commands/RegexIsMatchCommand.cs

```diff
@@ -0,0 +1,91 @@
+    {
+        [XmlAttribute]
+        [Attributes.PropertyAttributes.PropertyDescription("Please input the data you want to perform regex on.")]
+        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        [Attributes.PropertyAttributes.InputSpecification("Enter Variable or Text to apply Regex on.")]
+        [Attributes.PropertyAttributes.SampleUsage("**Hello** or **vSomeVariable**")]
+        [Attributes.PropertyAttributes.Remarks("")]
+        public string v_InputTextData { get; set; }
+
+        [XmlAttribute]
+        [Attributes.PropertyAttributes.PropertyDescription("Enter Regex Pattern.")]
+        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        [Attributes.PropertyAttributes.InputSpecification("Enter a Regex Pattern to apply to check if a match exists.")]
+        [Attributes.PropertyAttributes.SampleUsage(@"^([\w\-]+)")]
+        [Attributes.PropertyAttributes.Remarks("")]
+        public string v_RegEx { get; set; }
+
+        [XmlAttribute]
+        [Attributes.PropertyAttributes.PropertyDescription("Please enter the variable to get Regex result.")]
+        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        [Attributes.PropertyAttributes.InputSpecification("Enter a Variable to get result.")]
+        [Attributes.PropertyAttributes.SampleUsage("**vSomeVariable**")]
+        [Attributes.PropertyAttributes.Remarks("")]
+        public string v_ResultVariable { get; set; }
+
+        public RegexIsMatchCommand()
+        {
+            this.CommandName = "RegexIsMatchCommand";
+            this.SelectionName = "Regex IsMatch";
+            this.CommandEnabled = true;
+            this.CustomRendering = true;
+        }
+
+        public override void RunCommand(object sender)
+        {
+            var engine = (Core.Automation.Engine.AutomationEngineInstance)sender;
+            var vInputData = v_InputTextData.ConvertToUserVariable(engine);
+            string vRegex = v_RegEx.ConvertToUserVariable(engine);
+            bool isMatch = Regex.IsMatch(vInputData, vRegex);
+
+            Script.ScriptVariable resultVariable = new Script.ScriptVariable
+            {
+                VariableName = v_ResultVariable,
+                VariableValue = isMatch.ToString()
+            };
+
+            //Overwrites variable if it already exists
+            if (engine.VariableList.Exists(x => x.VariableName == resultVariable.VariableName))
+            {
+                Script.ScriptVariable temp = engine.VariableList.Where(x => x.VariableName == resultVariable.VariableName).FirstOrDefault();
+                engine.VariableList.Remove(temp);
+            }
+
+            engine.VariableList.Add(resultVariable);
```

Comment:
This method of adding variables to the variable list is only necessary for non-primitive types. You can replace all this with the StoreInUserVariable method when saving strings

Reply:
There's actually already `engine.AddVariable()` which accepts a script variable as an argument and checks to see if it exists before creating/updating it. We should use that to store any non-string values since the behavior of store/convert user variables is only designed to work with string values.

Reply:
These changes have been accommodated.